clang/include/clang/Basic/Attr.td

```diff
@@ -2648,10 +2648,11 @@ def SYCLIntelConstVar : SubsetSubject<Var,
                       [{S->getKind() != Decl::ImplicitParam &&
                         S->getKind() != Decl::ParmVar &&
                         S->getKind() != Decl::NonTypeTemplateParm &&
-                        (S->getType().isConstQualified() ||
+                        ((!S->hasAttr<SYCLDeviceGlobalAttr>() ||
```

Comment:
Question: why is this `!S->hasAttr<SYCLDeviceGlobalAttr>()`?  Shouldn't it be `S->hasAttr<SYCLDeviceGlobalAttr>()`?  i.e. why the negation?